src/gifwrite.c

```diff
@@ -12,15 +12,58 @@ void generate_GIF_data (struct context * context) {
+    findbounds:                                                                                                                  \
+    for (index = 0; index < framesize; index ++) if (buffer[index] != transparent) break;                                        \
+    if (index == framesize)                                                                                                      \
+      width = height = 1;                                                                                                        \
+    else {                                                                                                                       \
+      top = index / width;                                                                                                       \
+      height -= top;                                                                                                             \
+      for (index = 0; index < framesize; index ++) if (buffer[framesize - 1 - index] != transparent) break;                      \
+      height -= index / width;                                                                                                   \
+      for (left = 0; left < width; left ++) for (index = top; index < top + height; index ++)                                    \
+        if (buffer[index * context -> source -> width + left] != transparent) goto leftdone;                                     \
+      leftdone:                                                                                                                  \
+      width -= left;                                                                                                             \
+      uint_fast16_t col;                                                                                                         \
+      for (col = 0; col < width; col ++) for (index = top; index < top + height; index ++)                                       \
+        if (buffer[(index + 1) * context -> source -> width - 1 - col] != transparent) goto rightdone;                           \
+      rightdone:                                                                                                                 \
+      width -= col;                                                                                                              \
+    }                                                                                                                            \
+  }                                                                                                                              \
+  if (left || width != context -> source -> width) {                                                                             \
+    buffertype * target = buffer;                                                                                                \
```

Comment:
There's be no need for `buffertype` in C23, just do `typeof(buffer) target = buffer;`.